libimage/pull.go

```diff
@@ -592,6 +594,62 @@ func (r *Runtime) copySingleImageFromRegistry(ctx context.Context, imageName str
 		return nil
 	}
 
+	// Avoid running out of time when running inside a systemd unit by
+	// regularly increasing the timeout.
+	if socketPath, ok := os.LookupEnv("NOTIFY_SOCKET"); ok {
+		socketAddr := &net.UnixAddr{
+			Name: socketPath,
+			Net:  "unixgram",
+		}
+		conn, err := net.DialUnix(socketAddr.Net, nil, socketAddr)
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+
+		numExtensions := 10
+		extension := 30 * time.Second
+		timerFrequency := 25 * time.Second // Fire the timer at a higher frequency to avoid a race
+		timer := time.NewTimer(timerFrequency)
```

Comment:
We can simplify this by using time.NewTicker(), then we do not need the timer.Reset(extension) below.

Note that you still extend by 30s, thus we still can run into the race so this should be changed.

Reply:
Thanks for catching!